src/main/java/com/volasoftware/tinder/services/UserService.java

```diff
@@ -46,31 +71,22 @@ public void registerUser(UserDto userDto) throws MessagingException, IOException
         user.setEnabled(false);
         userRepository.saveAndFlush(user);
 
-        String uuid = UUID.randomUUID().toString();
-        Token token = new Token(
-                uuid,
-                LocalDateTime.now(),
-                LocalDateTime.now().plusDays(2),
-                user
-        );
+        Verification token = new Verification();
+        token.setUserId(user);
+        token.setToken(UUID.randomUUID().toString());
+        token.setCreatedDate(LocalDateTime.now());
+        token.setExpirationDate(LocalDateTime.now().plusDays(2));
+        verificationRepository.saveAndFlush(token);
 
-        tokenRepository.save(token);
-
-        sendMail();
-        //return token;
+        MimeMessage message = mailSender.createMimeMessage();
+        message.setFrom(new InternetAddress("ang3lkirilov@gmail.com"));
+        message.setRecipients(MimeMessage.RecipientType.TO,user.getEmail());
+        message.setSubject("Verification");
+        message.setContent(getEmailContent(token.getToken()),"text/html; charset=utf-8");
+        mailSender.send(message);
```

Comment:
This can be moved to a separate service class (EmailSenderService for example) to handle email sending only

Reply:
The EmailSenderService will be made when merging the next task.